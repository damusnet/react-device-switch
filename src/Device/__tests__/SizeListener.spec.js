import React from "react";
import { render, cleanup } from "react-testing-library";

import DeviceSizeListener from "../SizeListener";
import DeviceSwitch from "../Switch";

const BREAKPOINT = "min-width: 600px";

describe("<DeviceSizeListener />", () => {
  afterEach(cleanup);

  it("creates a provider correctly on the server", () => {
    const { container } = render(
      <DeviceSizeListener breakpoint={BREAKPOINT}>
        <DeviceSwitch
          desktopRender="Desktop Content"
          mobileRender="Mobile Content"
        />
      </DeviceSizeListener>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("creates a provider correctly on the client for a desktop", () => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn(() => ({
        matches: true,
        addListener: jest.fn(),
        removeListener: jest.fn()
      }))
    });

    const { container } = render(
      <DeviceSizeListener breakpoint={BREAKPOINT}>
        <DeviceSwitch
          desktopRender="Desktop Content"
          mobileRender="Mobile Content"
        />
      </DeviceSizeListener>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("creates a provider correctly on the client for a mobile", () => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn(() => ({
        matches: false,
        addListener: jest.fn(),
        removeListener: jest.fn()
      }))
    });

    const { container } = render(
      <DeviceSizeListener breakpoint={BREAKPOINT}>
        <DeviceSwitch
          desktopRender="Desktop Content"
          mobileRender="Mobile Content"
        />
      </DeviceSizeListener>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("removes the matchMedia listener on componentWillUnmount", () => {
    const removeListener = jest.fn();

    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn(() => ({
        matches: false,
        addListener: jest.fn(),
        removeListener
      }))
    });

    const { unmount } = render(
      <DeviceSizeListener breakpoint={BREAKPOINT}>content</DeviceSizeListener>
    );

    unmount();

    expect(removeListener).toHaveBeenCalled();
  });
});
