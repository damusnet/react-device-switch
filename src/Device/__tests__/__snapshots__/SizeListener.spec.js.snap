// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeviceSizeListener /> creates a provider correctly on the client for a desktop 1`] = `Desktop Content`;

exports[`<DeviceSizeListener /> creates a provider correctly on the client for a mobile 1`] = `Mobile Content`;

exports[`<DeviceSizeListener /> creates a provider correctly on the server 1`] = `Mobile Content`;
